File: dqops/src/main/java/com/dqops/checks/column/partitioned/text/ColumnTextDailyPartitionedChecksSpec.java
/*
 * Copyright © 2021 DQOps (support@dqops.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.dqops.checks.column.partitioned.text;

import com.dqops.checks.AbstractCheckCategorySpec;
import com.dqops.checks.CheckTarget;
import com.dqops.checks.CheckTimeScale;
import com.dqops.checks.CheckType;
import com.dqops.checks.column.checkspecs.text.*;
import com.dqops.connectors.DataTypeCategory;
import com.dqops.metadata.id.ChildHierarchyNodeFieldMap;
import com.dqops.metadata.id.ChildHierarchyNodeFieldMapImpl;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.EqualsAndHashCode;

import java.util.Objects;

/**
 * Container of text data quality partitioned checks on a column level that are checking at a daily partition level.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@EqualsAndHashCode(callSuper = true)
public class ColumnTextDailyPartitionedChecksSpec extends AbstractCheckCategorySpec {
    public static final ChildHierarchyNodeFieldMapImpl<ColumnTextDailyPartitionedChecksSpec> FIELDS = new ChildHierarchyNodeFieldMapImpl<>(AbstractCheckCategorySpec.FIELDS) {
        {
            put("daily_partition_text_min_length", o -> o.dailyPartitionTextMinLength);
            put("daily_partition_text_max_length", o -> o.dailyPartitionTextMaxLength);
            put("daily_partition_text_mean_length", o -> o.dailyPartitionTextMeanLength);
            put("daily_partition_text_length_below_min_length", o -> o.dailyPartitionTextLengthBelowMinLength);
            put("daily_partition_text_length_below_min_length_percent", o -> o.dailyPartitionTextLengthBelowMinLengthPercent);
            put("daily_partition_text_length_above_max_length", o -> o.dailyPartitionTextLengthAboveMaxLength);
            put("daily_partition_text_length_above_max_length_percent", o -> o.dailyPartitionTextLengthAboveMaxLengthPercent);
            put("daily_partition_text_length_in_range_percent", o -> o.dailyPartitionTextLengthInRangePercent);

            put("daily_partition_min_word_count", o -> o.dailyPartitionMinWordCount);
            put("daily_partition_max_word_count", o -> o.dailyPartitionMaxWordCount);
        }
    };


    @JsonPropertyDescription("This check finds the length of the shortest text in a column. Then, it verifies that the minimum length is within an accepted range. It detects that the shortest text is too short. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextMinLengthCheckSpec dailyPartitionTextMinLength;

    @JsonPropertyDescription("This check finds the length of the longest text in a column. Then, it verifies that the maximum length is within an accepted range. It detects that the texts are too long or not long enough. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextMaxLengthCheckSpec dailyPartitionTextMaxLength;

    @JsonPropertyDescription("Verifies that the mean (average) length of texts in a column is within an accepted range. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextMeanLengthCheckSpec dailyPartitionTextMeanLength;

    @JsonPropertyDescription("The check counts the number of text values in the column that is below the length defined by the user as a parameter. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextLengthBelowMinLengthCheckSpec dailyPartitionTextLengthBelowMinLength;

    @JsonPropertyDescription("The check measures the percentage of text values in the column that is below the length defined by the user as a parameter. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextLengthBelowMinLengthPercentCheckSpec dailyPartitionTextLengthBelowMinLengthPercent;

    @JsonPropertyDescription("The check counts the number of text values in the column that is above the length defined by the user as a parameter. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextLengthAboveMaxLengthCheckSpec dailyPartitionTextLengthAboveMaxLength;

    @JsonPropertyDescription("The check measures the percentage of text values in the column that is above the length defined by the user as a parameter. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextLengthAboveMaxLengthPercentCheckSpec dailyPartitionTextLengthAboveMaxLengthPercent;

    @JsonPropertyDescription("The check measures the percentage of those text values with length in the range provided by the user in the column. Analyzes every daily partition and creates a separate data quality check result with the time period value that identifies the daily partition.")
    private ColumnTextLengthInRangePercentCheckSpec dailyPartitionTextLengthInRangePercent;

    @JsonPropertyDescription("This check finds the lowest word count of text in a column. Then, it verifies that the minimum length is within an accepted range. It detects that the text contains too less words.")
    private ColumnTextMinWordCountCheckSpec dailyPartitionMinWordCount;

    @JsonPropertyDescription("This check finds the highest word count of text in a column. Then, it verifies that the maximum length is within an accepted range. It detects that the text contains too many words.")
    private ColumnTextMaxWordCountCheckSpec dailyPartitionMaxWordCount;

    /**
     * Returns a minimum string length above check.
     * @return Minimum string length above check.
     */
    public ColumnTextMinLengthCheckSpec getDailyPartitionTextMinLength() {
        return dailyPartitionTextMinLength;
    }

    /**
     * Sets a new definition of a minimum string length check.
     * @param dailyPartitionTextMinLength Minimum string length check.
     */
    public void setDailyPartitionTextMinLength(ColumnTextMinLengthCheckSpec dailyPartitionTextMinLength) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextMinLength, dailyPartitionTextMinLength));
        this.dailyPartitionTextMinLength = dailyPartitionTextMinLength;
        propagateHierarchyIdToField(dailyPartitionTextMinLength, "daily_partition_text_min_length");
    }

    /**
     * Returns a maximum string length below check.
     * @return Maximum string length below check.
     */
    public ColumnTextMaxLengthCheckSpec getDailyPartitionTextMaxLength() {
        return dailyPartitionTextMaxLength;
    }

    /**
     * Sets a new definition of a maximum string length check.
     * @param dailyPartitionTextMaxLength Maximum string length check.
     */
    public void setDailyPartitionTextMaxLength(ColumnTextMaxLengthCheckSpec dailyPartitionTextMaxLength) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextMaxLength, dailyPartitionTextMaxLength));
        this.dailyPartitionTextMaxLength = dailyPartitionTextMaxLength;
        propagateHierarchyIdToField(dailyPartitionTextMaxLength, "daily_partition_text_max_length");
    }

    /**
     * Returns a mean string length between check.
     * @return Mean string length between check.
     */
    public ColumnTextMeanLengthCheckSpec getDailyPartitionTextMeanLength() {
        return dailyPartitionTextMeanLength;
    }

    /**
     * Sets a new definition of a mean string length check.
     * @param dailyPartitionTextMeanLength Mean string length check.
     */
    public void setDailyPartitionTextMeanLength(ColumnTextMeanLengthCheckSpec dailyPartitionTextMeanLength) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextMeanLength, dailyPartitionTextMeanLength));
        this.dailyPartitionTextMeanLength = dailyPartitionTextMeanLength;
        propagateHierarchyIdToField(dailyPartitionTextMeanLength, "daily_partition_text_mean_length");
    }

    /**
     * Returns a string length below min length count check.
     * @return String length below min length count check.
     */
    public ColumnTextLengthBelowMinLengthCheckSpec getDailyPartitionTextLengthBelowMinLength() {
        return dailyPartitionTextLengthBelowMinLength;
    }

    /**
     * Sets a new definition of a string length below min length count check.
     * @param dailyPartitionTextLengthBelowMinLength String length below min length count check.
     */
    public void setDailyPartitionTextLengthBelowMinLength(ColumnTextLengthBelowMinLengthCheckSpec dailyPartitionTextLengthBelowMinLength) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextLengthBelowMinLength, dailyPartitionTextLengthBelowMinLength));
        this.dailyPartitionTextLengthBelowMinLength = dailyPartitionTextLengthBelowMinLength;
        propagateHierarchyIdToField(dailyPartitionTextLengthBelowMinLength, "daily_partition_text_length_below_min_length");
    }

    /**
     * Returns a string length below min length percent check.
     * @return String length below min length percent check.
     */
    public ColumnTextLengthBelowMinLengthPercentCheckSpec getDailyPartitionTextLengthBelowMinLengthPercent() {
        return dailyPartitionTextLengthBelowMinLengthPercent;
    }

    /**
     * Sets a new definition of a string length below min length percent check.
     * @param dailyPartitionTextLengthBelowMinLengthPercent String length below min length percent check.
     */
    public void setDailyPartitionTextLengthBelowMinLengthPercent(ColumnTextLengthBelowMinLengthPercentCheckSpec dailyPartitionTextLengthBelowMinLengthPercent) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextLengthBelowMinLengthPercent, dailyPartitionTextLengthBelowMinLengthPercent));
        this.dailyPartitionTextLengthBelowMinLengthPercent = dailyPartitionTextLengthBelowMinLengthPercent;
        propagateHierarchyIdToField(dailyPartitionTextLengthBelowMinLengthPercent, "daily_partition_text_length_below_min_length_percent");
    }

    /**
     * Returns a string length above max length count check.
     * @return String length above max length count check.
     */
    public ColumnTextLengthAboveMaxLengthCheckSpec getDailyPartitionTextLengthAboveMaxLength() {
        return dailyPartitionTextLengthAboveMaxLength;
    }

    /**
     * Sets a new definition of a string length above max length count check.
     * @param dailyPartitionTextLengthAboveMaxLength String length above max length count check.
     */
    public void setDailyPartitionTextLengthAboveMaxLength(ColumnTextLengthAboveMaxLengthCheckSpec dailyPartitionTextLengthAboveMaxLength) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextLengthAboveMaxLength, dailyPartitionTextLengthAboveMaxLength));
        this.dailyPartitionTextLengthAboveMaxLength = dailyPartitionTextLengthAboveMaxLength;
        propagateHierarchyIdToField(dailyPartitionTextLengthAboveMaxLength, "daily_partition_text_length_above_max_length");
    }

    /**
     * Returns a string length above max length percent check.
     * @return String length above max length percent check.
     */
    public ColumnTextLengthAboveMaxLengthPercentCheckSpec getDailyPartitionTextLengthAboveMaxLengthPercent() {
        return dailyPartitionTextLengthAboveMaxLengthPercent;
    }

    /**
     * Sets a new definition of a string length above max length percent check.
     * @param dailyPartitionTextLengthAboveMaxLengthPercent String length above max length percent check.
     */
    public void setDailyPartitionTextLengthAboveMaxLengthPercent(ColumnTextLengthAboveMaxLengthPercentCheckSpec dailyPartitionTextLengthAboveMaxLengthPercent) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextLengthAboveMaxLengthPercent, dailyPartitionTextLengthAboveMaxLengthPercent));
        this.dailyPartitionTextLengthAboveMaxLengthPercent = dailyPartitionTextLengthAboveMaxLengthPercent;
        propagateHierarchyIdToField(dailyPartitionTextLengthAboveMaxLengthPercent, "daily_partition_text_length_above_max_length_percent");
    }

    /**
     * Returns a string length in range percent check.
     * @return String length in range percent check.
     */
    public ColumnTextLengthInRangePercentCheckSpec getDailyPartitionTextLengthInRangePercent() {
        return dailyPartitionTextLengthInRangePercent;
    }

    /**
     * Sets a new definition of a string length in range percent check.
     * @param dailyPartitionTextLengthInRangePercent String length in range percent check.
     */
    public void setDailyPartitionTextLengthInRangePercent(ColumnTextLengthInRangePercentCheckSpec dailyPartitionTextLengthInRangePercent) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionTextLengthInRangePercent, dailyPartitionTextLengthInRangePercent));
        this.dailyPartitionTextLengthInRangePercent = dailyPartitionTextLengthInRangePercent;
        propagateHierarchyIdToField(dailyPartitionTextLengthInRangePercent, "daily_partition_text_length_in_range_percent");
    }

    /**
     * Returns a text min word count percent check.
     * @return Text min word count percent check.
     */
    public ColumnTextMinWordCountCheckSpec getDailyPartitionMinWordCount() {
        return dailyPartitionMinWordCount;
    }

    /**
     * Sets a new definition of a text min word count percent check.
     * @param dailyPartitionMinWordCount Text min word count percent check.
     */
    public void setDailyPartitionMinWordCount(ColumnTextMinWordCountCheckSpec dailyPartitionMinWordCount) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionMinWordCount, dailyPartitionMinWordCount));
        this.dailyPartitionMinWordCount = dailyPartitionMinWordCount;
        propagateHierarchyIdToField(dailyPartitionMinWordCount, "daily_partition_min_word_count");
    }

    /**
     * Returns a text max word count percent check.
     * @return Text max word count percent check.
     */
    public ColumnTextMaxWordCountCheckSpec getDailyPartitionMaxWordCount() {
        return dailyPartitionMaxWordCount;
    }

    /**
     * Sets a new definition of a text max word count percent check.
     * @param dailyPartitionMaxWordCount Text max word count percent check.
     */
    public void setDailyPartitionMaxWordCount(ColumnTextMaxWordCountCheckSpec dailyPartitionMaxWordCount) {
        this.setDirtyIf(!Objects.equals(this.dailyPartitionMaxWordCount, dailyPartitionMaxWordCount));
        this.dailyPartitionMaxWordCount = dailyPartitionMaxWordCount;
        propagateHierarchyIdToField(dailyPartitionMaxWordCount, "daily_partition_max_word_count");
    }


    /**
     * Returns the child map on the spec class with all fields.
     *
     * @return Return the field map.
     */
    @Override
    protected ChildHierarchyNodeFieldMap getChildMap() {
        return FIELDS;
    }

    /**
     * Gets the check target appropriate for all checks in this category.
     *
     * @return Corresponding check target.
     */
    @Override
    @JsonIgnore
    public CheckTarget getCheckTarget() {
        return CheckTarget.column;
    }

    /**
     * Gets the check type appropriate for all checks in this category.
     *
     * @return Corresponding check type.
     */
    @Override
    @JsonIgnore
    public CheckType getCheckType() {
        return CheckType.partitioned;
    }

    /**
     * Gets the check timescale appropriate for all checks in this category.
     *
     * @return Corresponding check timescale.
     */
    @Override
    @JsonIgnore
    public CheckTimeScale getCheckTimeScale() {
        return CheckTimeScale.daily;
    }

    /**
     * Returns an array of supported data type categories. DQOps uses this list when activating default data quality checks.
     *
     * @return Array of supported data type categories.
     */
    @Override
    @JsonIgnore
    public DataTypeCategory[] getSupportedDataTypeCategories() {
        return DataTypeCategory.STRING;
    }
}

File: distribution/python/dqops/client/models/column_text_daily_partitioned_checks_spec.py
from typing import TYPE_CHECKING, Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

if TYPE_CHECKING:
    from ..models.column_text_daily_partitioned_checks_spec_custom_checks import (
        ColumnTextDailyPartitionedChecksSpecCustomChecks,
    )
    from ..models.column_text_length_above_max_length_check_spec import (
        ColumnTextLengthAboveMaxLengthCheckSpec,
    )
    from ..models.column_text_length_above_max_length_percent_check_spec import (
        ColumnTextLengthAboveMaxLengthPercentCheckSpec,
    )
    from ..models.column_text_length_below_min_length_check_spec import (
        ColumnTextLengthBelowMinLengthCheckSpec,
    )
    from ..models.column_text_length_below_min_length_percent_check_spec import (
        ColumnTextLengthBelowMinLengthPercentCheckSpec,
    )
    from ..models.column_text_length_in_range_percent_check_spec import (
        ColumnTextLengthInRangePercentCheckSpec,
    )
    from ..models.column_text_max_length_check_spec import ColumnTextMaxLengthCheckSpec
    from ..models.column_text_max_word_count_check_spec import (
        ColumnTextMaxWordCountCheckSpec,
    )
    from ..models.column_text_mean_length_check_spec import (
        ColumnTextMeanLengthCheckSpec,
    )
    from ..models.column_text_min_length_check_spec import ColumnTextMinLengthCheckSpec
    from ..models.column_text_min_word_count_check_spec import (
        ColumnTextMinWordCountCheckSpec,
    )


T = TypeVar("T", bound="ColumnTextDailyPartitionedChecksSpec")


@_attrs_define
class ColumnTextDailyPartitionedChecksSpec:
    """
    Attributes:
        custom_checks (Union[Unset, ColumnTextDailyPartitionedChecksSpecCustomChecks]): Dictionary of additional custom
            checks within this category. The keys are check names defined in the definition section. The sensor parameters
            and rules should match the type of the configured sensor and rule for the custom check.
        daily_partition_text_min_length (Union[Unset, ColumnTextMinLengthCheckSpec]):
        daily_partition_text_max_length (Union[Unset, ColumnTextMaxLengthCheckSpec]):
        daily_partition_text_mean_length (Union[Unset, ColumnTextMeanLengthCheckSpec]):
        daily_partition_text_length_below_min_length (Union[Unset, ColumnTextLengthBelowMinLengthCheckSpec]):
        daily_partition_text_length_below_min_length_percent (Union[Unset,
            ColumnTextLengthBelowMinLengthPercentCheckSpec]):
        daily_partition_text_length_above_max_length (Union[Unset, ColumnTextLengthAboveMaxLengthCheckSpec]):
        daily_partition_text_length_above_max_length_percent (Union[Unset,
            ColumnTextLengthAboveMaxLengthPercentCheckSpec]):
        daily_partition_text_length_in_range_percent (Union[Unset, ColumnTextLengthInRangePercentCheckSpec]):
        daily_partition_min_word_count (Union[Unset, ColumnTextMinWordCountCheckSpec]):
        daily_partition_max_word_count (Union[Unset, ColumnTextMaxWordCountCheckSpec]):
    """

    custom_checks: Union[Unset, "ColumnTextDailyPartitionedChecksSpecCustomChecks"] = (
        UNSET
    )
    daily_partition_text_min_length: Union[Unset, "ColumnTextMinLengthCheckSpec"] = (
        UNSET
    )
    daily_partition_text_max_length: Union[Unset, "ColumnTextMaxLengthCheckSpec"] = (
        UNSET
    )
    daily_partition_text_mean_length: Union[Unset, "ColumnTextMeanLengthCheckSpec"] = (
        UNSET
    )
    daily_partition_text_length_below_min_length: Union[
        Unset, "ColumnTextLengthBelowMinLengthCheckSpec"
    ] = UNSET
    daily_partition_text_length_below_min_length_percent: Union[
        Unset, "ColumnTextLengthBelowMinLengthPercentCheckSpec"
    ] = UNSET
    daily_partition_text_length_above_max_length: Union[
        Unset, "ColumnTextLengthAboveMaxLengthCheckSpec"
    ] = UNSET
    daily_partition_text_length_above_max_length_percent: Union[
        Unset, "ColumnTextLengthAboveMaxLengthPercentCheckSpec"
    ] = UNSET
    daily_partition_text_length_in_range_percent: Union[
        Unset, "ColumnTextLengthInRangePercentCheckSpec"
    ] = UNSET
    daily_partition_min_word_count: Union[Unset, "ColumnTextMinWordCountCheckSpec"] = (
        UNSET
    )
    daily_partition_max_word_count: Union[Unset, "ColumnTextMaxWordCountCheckSpec"] = (
        UNSET
    )
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        custom_checks: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.custom_checks, Unset):
            custom_checks = self.custom_checks.to_dict()

        daily_partition_text_min_length: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.daily_partition_text_min_length, Unset):
            daily_partition_text_min_length = (
                self.daily_partition_text_min_length.to_dict()
            )

        daily_partition_text_max_length: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.daily_partition_text_max_length, Unset):
            daily_partition_text_max_length = (
                self.daily_partition_text_max_length.to_dict()
            )

        daily_partition_text_mean_length: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.daily_partition_text_mean_length, Unset):
            daily_partition_text_mean_length = (
                self.daily_partition_text_mean_length.to_dict()
            )

        daily_partition_text_length_below_min_length: Union[Unset, Dict[str, Any]] = (
            UNSET
        )
        if not isinstance(self.daily_partition_text_length_below_min_length, Unset):
            daily_partition_text_length_below_min_length = (
                self.daily_partition_text_length_below_min_length.to_dict()
            )

        daily_partition_text_length_below_min_length_percent: Union[
            Unset, Dict[str, Any]
        ] = UNSET
        if not isinstance(
            self.daily_partition_text_length_below_min_length_percent, Unset
        ):
            daily_partition_text_length_below_min_length_percent = (
                self.daily_partition_text_length_below_min_length_percent.to_dict()
            )

        daily_partition_text_length_above_max_length: Union[Unset, Dict[str, Any]] = (
            UNSET
        )
        if not isinstance(self.daily_partition_text_length_above_max_length, Unset):
            daily_partition_text_length_above_max_length = (
                self.daily_partition_text_length_above_max_length.to_dict()
            )

        daily_partition_text_length_above_max_length_percent: Union[
            Unset, Dict[str, Any]
        ] = UNSET
        if not isinstance(
            self.daily_partition_text_length_above_max_length_percent, Unset
        ):
            daily_partition_text_length_above_max_length_percent = (
                self.daily_partition_text_length_above_max_length_percent.to_dict()
            )

        daily_partition_text_length_in_range_percent: Union[Unset, Dict[str, Any]] = (
            UNSET
        )
        if not isinstance(self.daily_partition_text_length_in_range_percent, Unset):
            daily_partition_text_length_in_range_percent = (
                self.daily_partition_text_length_in_range_percent.to_dict()
            )

        daily_partition_min_word_count: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.daily_partition_min_word_count, Unset):
            daily_partition_min_word_count = (
                self.daily_partition_min_word_count.to_dict()
            )

        daily_partition_max_word_count: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.daily_partition_max_word_count, Unset):
            daily_partition_max_word_count = (
                self.daily_partition_max_word_count.to_dict()
            )

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if custom_checks is not UNSET:
            field_dict["custom_checks"] = custom_checks
        if daily_partition_text_min_length is not UNSET:
            field_dict["daily_partition_text_min_length"] = (
                daily_partition_text_min_length
            )
        if daily_partition_text_max_length is not UNSET:
            field_dict["daily_partition_text_max_length"] = (
                daily_partition_text_max_length
            )
        if daily_partition_text_mean_length is not UNSET:
            field_dict["daily_partition_text_mean_length"] = (
                daily_partition_text_mean_length
            )
        if daily_partition_text_length_below_min_length is not UNSET:
            field_dict["daily_partition_text_length_below_min_length"] = (
                daily_partition_text_length_below_min_length
            )
        if daily_partition_text_length_below_min_length_percent is not UNSET:
            field_dict["daily_partition_text_length_below_min_length_percent"] = (
                daily_partition_text_length_below_min_length_percent
            )
        if daily_partition_text_length_above_max_length is not UNSET:
            field_dict["daily_partition_text_length_above_max_length"] = (
                daily_partition_text_length_above_max_length
            )
        if daily_partition_text_length_above_max_length_percent is not UNSET:
            field_dict["daily_partition_text_length_above_max_length_percent"] = (
                daily_partition_text_length_above_max_length_percent
            )
        if daily_partition_text_length_in_range_percent is not UNSET:
            field_dict["daily_partition_text_length_in_range_percent"] = (
                daily_partition_text_length_in_range_percent
            )
        if daily_partition_min_word_count is not UNSET:
            field_dict["daily_partition_min_word_count"] = (
                daily_partition_min_word_count
            )
        if daily_partition_max_word_count is not UNSET:
            field_dict["daily_partition_max_word_count"] = (
                daily_partition_max_word_count
            )

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        from ..models.column_text_daily_partitioned_checks_spec_custom_checks import (
            ColumnTextDailyPartitionedChecksSpecCustomChecks,
        )
        from ..models.column_text_length_above_max_length_check_spec import (
            ColumnTextLengthAboveMaxLengthCheckSpec,
        )
        from ..models.column_text_length_above_max_length_percent_check_spec import (
            ColumnTextLengthAboveMaxLengthPercentCheckSpec,
        )
        from ..models.column_text_length_below_min_length_check_spec import (
            ColumnTextLengthBelowMinLengthCheckSpec,
        )
        from ..models.column_text_length_below_min_length_percent_check_spec import (
            ColumnTextLengthBelowMinLengthPercentCheckSpec,
        )
        from ..models.column_text_length_in_range_percent_check_spec import (
            ColumnTextLengthInRangePercentCheckSpec,
        )
        from ..models.column_text_max_length_check_spec import (
            ColumnTextMaxLengthCheckSpec,
        )
        from ..models.column_text_max_word_count_check_spec import (
            ColumnTextMaxWordCountCheckSpec,
        )
        from ..models.column_text_mean_length_check_spec import (
            ColumnTextMeanLengthCheckSpec,
        )
        from ..models.column_text_min_length_check_spec import (
            ColumnTextMinLengthCheckSpec,
        )
        from ..models.column_text_min_word_count_check_spec import (
            ColumnTextMinWordCountCheckSpec,
        )

        d = src_dict.copy()
        _custom_checks = d.pop("custom_checks", UNSET)
        custom_checks: Union[Unset, ColumnTextDailyPartitionedChecksSpecCustomChecks]
        if isinstance(_custom_checks, Unset):
            custom_checks = UNSET
        else:
            custom_checks = ColumnTextDailyPartitionedChecksSpecCustomChecks.from_dict(
                _custom_checks
            )

        _daily_partition_text_min_length = d.pop(
            "daily_partition_text_min_length", UNSET
        )
        daily_partition_text_min_length: Union[Unset, ColumnTextMinLengthCheckSpec]
        if isinstance(_daily_partition_text_min_length, Unset):
            daily_partition_text_min_length = UNSET
        else:
            daily_partition_text_min_length = ColumnTextMinLengthCheckSpec.from_dict(
                _daily_partition_text_min_length
            )

        _daily_partition_text_max_length = d.pop(
            "daily_partition_text_max_length", UNSET
        )
        daily_partition_text_max_length: Union[Unset, ColumnTextMaxLengthCheckSpec]
        if isinstance(_daily_partition_text_max_length, Unset):
            daily_partition_text_max_length = UNSET
        else:
            daily_partition_text_max_length = ColumnTextMaxLengthCheckSpec.from_dict(
                _daily_partition_text_max_length
            )

        _daily_partition_text_mean_length = d.pop(
            "daily_partition_text_mean_length", UNSET
        )
        daily_partition_text_mean_length: Union[Unset, ColumnTextMeanLengthCheckSpec]
        if isinstance(_daily_partition_text_mean_length, Unset):
            daily_partition_text_mean_length = UNSET
        else:
            daily_partition_text_mean_length = ColumnTextMeanLengthCheckSpec.from_dict(
                _daily_partition_text_mean_length
            )

        _daily_partition_text_length_below_min_length = d.pop(
            "daily_partition_text_length_below_min_length", UNSET
        )
        daily_partition_text_length_below_min_length: Union[
            Unset, ColumnTextLengthBelowMinLengthCheckSpec
        ]
        if isinstance(_daily_partition_text_length_below_min_length, Unset):
            daily_partition_text_length_below_min_length = UNSET
        else:
            daily_partition_text_length_below_min_length = (
                ColumnTextLengthBelowMinLengthCheckSpec.from_dict(
                    _daily_partition_text_length_below_min_length
                )
            )

        _daily_partition_text_length_below_min_length_percent = d.pop(
            "daily_partition_text_length_below_min_length_percent", UNSET
        )
        daily_partition_text_length_below_min_length_percent: Union[
            Unset, ColumnTextLengthBelowMinLengthPercentCheckSpec
        ]
        if isinstance(_daily_partition_text_length_below_min_length_percent, Unset):
            daily_partition_text_length_below_min_length_percent = UNSET
        else:
            daily_partition_text_length_below_min_length_percent = (
                ColumnTextLengthBelowMinLengthPercentCheckSpec.from_dict(
                    _daily_partition_text_length_below_min_length_percent
                )
            )

        _daily_partition_text_length_above_max_length = d.pop(
            "daily_partition_text_length_above_max_length", UNSET
        )
        daily_partition_text_length_above_max_length: Union[
            Unset, ColumnTextLengthAboveMaxLengthCheckSpec
        ]
        if isinstance(_daily_partition_text_length_above_max_length, Unset):
            daily_partition_text_length_above_max_length = UNSET
        else:
            daily_partition_text_length_above_max_length = (
                ColumnTextLengthAboveMaxLengthCheckSpec.from_dict(
                    _daily_partition_text_length_above_max_length
                )
            )

        _daily_partition_text_length_above_max_length_percent = d.pop(
            "daily_partition_text_length_above_max_length_percent", UNSET
        )
        daily_partition_text_length_above_max_length_percent: Union[
            Unset, ColumnTextLengthAboveMaxLengthPercentCheckSpec
        ]
        if isinstance(_daily_partition_text_length_above_max_length_percent, Unset):
            daily_partition_text_length_above_max_length_percent = UNSET
        else:
            daily_partition_text_length_above_max_length_percent = (
                ColumnTextLengthAboveMaxLengthPercentCheckSpec.from_dict(
                    _daily_partition_text_length_above_max_length_percent
                )
            )

        _daily_partition_text_length_in_range_percent = d.pop(
            "daily_partition_text_length_in_range_percent", UNSET
        )
        daily_partition_text_length_in_range_percent: Union[
            Unset, ColumnTextLengthInRangePercentCheckSpec
        ]
        if isinstance(_daily_partition_text_length_in_range_percent, Unset):
            daily_partition_text_length_in_range_percent = UNSET
        else:
            daily_partition_text_length_in_range_percent = (
                ColumnTextLengthInRangePercentCheckSpec.from_dict(
                    _daily_partition_text_length_in_range_percent
                )
            )

        _daily_partition_min_word_count = d.pop("daily_partition_min_word_count", UNSET)
        daily_partition_min_word_count: Union[Unset, ColumnTextMinWordCountCheckSpec]
        if isinstance(_daily_partition_min_word_count, Unset):
            daily_partition_min_word_count = UNSET
        else:
            daily_partition_min_word_count = ColumnTextMinWordCountCheckSpec.from_dict(
                _daily_partition_min_word_count
            )

        _daily_partition_max_word_count = d.pop("daily_partition_max_word_count", UNSET)
        daily_partition_max_word_count: Union[Unset, ColumnTextMaxWordCountCheckSpec]
        if isinstance(_daily_partition_max_word_count, Unset):
            daily_partition_max_word_count = UNSET
        else:
            daily_partition_max_word_count = ColumnTextMaxWordCountCheckSpec.from_dict(
                _daily_partition_max_word_count
            )

        column_text_daily_partitioned_checks_spec = cls(
            custom_checks=custom_checks,
            daily_partition_text_min_length=daily_partition_text_min_length,
            daily_partition_text_max_length=daily_partition_text_max_length,
            daily_partition_text_mean_length=daily_partition_text_mean_length,
            daily_partition_text_length_below_min_length=daily_partition_text_length_below_min_length,
            daily_partition_text_length_below_min_length_percent=daily_partition_text_length_below_min_length_percent,
            daily_partition_text_length_above_max_length=daily_partition_text_length_above_max_length,
            daily_partition_text_length_above_max_length_percent=daily_partition_text_length_above_max_length_percent,
            daily_partition_text_length_in_range_percent=daily_partition_text_length_in_range_percent,
            daily_partition_min_word_count=daily_partition_min_word_count,
            daily_partition_max_word_count=daily_partition_max_word_count,
        )

        column_text_daily_partitioned_checks_spec.additional_properties = d
        return column_text_daily_partitioned_checks_spec

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties
